exploration/0483-beauty-contest.md

```diff
@@ -0,0 +1,129 @@
+# Syntax Variation Beauty Contest
+
+As discussed in the 2023-09-25 teleconference, we need to choose a syntax to proceed with.
+This page hosts various options for considering in the 2023-10-02 call.
+
+## Current
+
+This is the current syntax.
+The list of messages in this section also serves as the basis for all other examples.
+
+```
+{Hello world!}
+
+{Hello {$user}}
+
+input $var :function option=value
+{Hello {$var}}
+
+input $var :function option=value
+local $foo = {$bar :function option=value}
+{Hello {$var}, you have a {$foo}}
+
+match {$foo} {$bar}
+when foo bar {Hello {$foo} you have a {$var}}
+when * * {{$foo} hello you have a {$var}}
+
+match {$foo :function option=value} {$bar :function option=value}
+when a b {  {$foo} is {$bar}  }
+when x y {  {$foo} is {$bar}  }
+when * * {  {$foo} is {$bar}  }
+
+input $var :function option=value local $foo = {$bar :function option=value}{Hello {$var}, you have a {$foo}}
+
+match {$foo} {$bar} when foo bar {Hello {$foo} you have a {$var}} when * * {{$foo} hello you have a {$var}}
+
+match {$foo :function option=value} {$bar :function option=value}when a b {  {$foo} is {$bar}  }when x y {  {$foo} is {$bar}  }when * * {  {$foo} is {$bar}  }
+```
+
+## Invert for Text Mode
+
+Consumes exterior whitespace.
+
+```
+Hello world!
+
+Hello {$user}
+
+{input $var :function option=value}
+Hello {$var}
+
+{input $var :function option=value}
+{local $foo = $bar :function option=value}
+Hello {$var}, you have a {$foo}
+
+{match {$foo} {$bar}}
+{when foo bar} Hello {$foo} you have a {$var}
+{when * *} {$foo} hello you have a {$var}
+
+{match {$foo :function option=value} {$bar :function option=value}}
+{when a b} {  {$foo} is {$bar}  }
+{when x y} {  {$foo} is {$bar}  }
+{when * *} {|  |}{$foo} is {$bar}{|  |}
+
+{input $var :function option=value}{local $foo = $bar :function option=value}Hello {$var}, you have a {$foo}
+
+{match {$foo} {$bar}}{when foo bar} Hello {$foo} you have a {$var}{when * *} {$foo} hello you have a {$var}
+
+{match {$foo :function option=value}{$bar :function option=value}}{when a b} {  {$foo} is {$bar}  }{when x y} {  {$foo} is {$bar}  }{when * *} {|  |}{$foo} is {$bar}{|  |}
+```
+
+## Use sigils for code mode
+
+Try to redues the use of `{`/`}` to just expressions and placeholders instead of the three
+uses we have now (the other use is for patterns). This requires escaping whitespace or using
+a placeholder for it.
+
+The sigil `#` was chosen because `#define` type constructs are fairly common. Introduces `[`/`]` for keys.
+
+```
+#{input $var :function option=value}
+Hello {$var}
+
+#{input $var :function option=value}
```

Comment:
For consistency with `#local` and `#match`, this could be:

```suggestion
#input {$var :function option=value}
Hello {$var}

#input {$var :function option=value}
```

Reply:
(Same for the single-line version.)